project3/coordinator/messenger.h

```diff
@@ -28,23 +29,17 @@ class Messenger {
    * @param msg_log The coordinator's message log.
    * @param participant The specific participant that this Messenger
    *    communicates with.
-   * @param connected_participants The coordinator's set of connected
-   * participants.
    */
-  Messenger(std::shared_ptr<queue::Queue<MessageLog::Message>> msg_queue,
-            std::shared_ptr<MessageLog> msg_log,
-            std::shared_ptr<ConnectedParticipants::Participant> participant,
-            std::shared_ptr<ConnectedParticipants> connected_participants);
+  Messenger(std::shared_ptr<MessageLog> msg_log,
+            ConnectedParticipants::Participant participant);
 
   /**
-   * @brief Broadcasts a given message to all active participants.
-   * @note Messages sent by the messenger should be pushed in the message queue before
-   *       this function is called. Guarantees that messages will be broadcast in the order
-   *       they were received.
+   * @brief Sends a given message to this messenger's participant.
+   * @param msg The message to send.
+   * @param missed_msg if the message being sent is a missed message.
    * @return true on success, false on failure
-   *
    */
-  bool BroadcastMessage();
+  bool SendMessage(MessageLog::Message msg, bool missed_msg);
```

Comment:
Maybe the first argument should be a `const` reference? If you're actually modifying it, it really should be a pointer in accordance with the style guide. Also, perhaps a better way to do what is to remove the missed-message-specific stuff into the `SendMissedMessages` function so you don't have to have that second argument.

Reply:
So, to keep with the thread-safety, `SendMissedMessages()` calls  `SendMessage()`, and in `SendMessage()`, we update the timestamp of the message and add it to the message log. For this reason, I think I need to keep that second argument. 
I suppose I could just create a separate function in `UpdateMessage()` that does this and have the coordinator call that after sending a message,  but I'm not sure that that is a better design choice?

Reply:
Hmm, is this really a thread-safety issue though? As far as I can tell, the only state that's getting updated there is the message timestamp, and I doubt the same message would be getting sent concurrently from multiple threads... Also, it's being added to the log, but that operation is already thread-safe. Really, the only reason I can see for doing it this way is if you needed the sending, timestamp-updating, and log-updating operations to operate as a single atomic block. Is there something I'm missing that necessitates that?

Reply:
Yeah, the timestamp thread-safety issue is a highly unlikely edge case. I'll change it.